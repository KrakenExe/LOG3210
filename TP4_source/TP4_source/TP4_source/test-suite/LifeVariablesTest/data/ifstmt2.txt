Declaration
bool a;
bool b;
bool c;
EndDeclaration

if(a) {
    b = a || c;
}
else {
    b = a;
}
a = c;