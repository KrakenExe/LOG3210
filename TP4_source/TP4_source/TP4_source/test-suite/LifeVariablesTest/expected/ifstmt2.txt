===== STEP 0 ===== 
REF : [a]
DEF : []
IN  : [a, c]
OUT : [a, c]
SUCC: [_step1, _step2]
PRED: []

===== STEP 1 ===== 
REF : [a, c]
DEF : [b]
IN  : [a, c]
OUT : [c]
SUCC: [_step3]
PRED: [_step0]

===== STEP 2 ===== 
REF : [a]
DEF : [b]
IN  : [a, c]
OUT : [c]
SUCC: [_step3]
PRED: [_step0]

===== STEP 3 ===== 
REF : [c]
DEF : [a]
IN  : [c]
OUT : []
SUCC: []
PRED: [_step1, _step2]

